tests/e2e/lib/backup.go

```diff
@@ -145,3 +145,50 @@ func BackupErrorLogs(c *kubernetes.Clientset, ocClient client.Client, namespace
 	bl := BackupLogs(c, ocClient, namespace, name)
 	return errorLogsExcludingIgnored(bl)
 }
+
+func GetBackupRepositoryList(c client.Client, namespace string) (*velero.BackupRepositoryList, error) {
+	backupRepositoryList := &velero.BackupRepositoryList{
+		Items: []velero.BackupRepository{},
+	}
+	err := c.List(context.Background(), backupRepositoryList, client.InNamespace(namespace))
+	// if there is an error, just print, don't return the error
+	if err != nil {
+		log.Printf("error getting BackupRepository list: %v", err)
+	}
+	return backupRepositoryList, nil
+}
+
+func DeleteBackupRepository(c client.Client, namespace string, name string) error {
+	backupRepository := &velero.BackupRepository{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      name,
+		},
+	}
+	err := c.Delete(context.Background(), backupRepository)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// DeleteBackupRepositoryByRegex deletes a BackupRepository instance that matches the given regex pattern.
+func DeleteBackupRepositoryByRegex(c client.Client, namespace string) error {
+	log.Printf("Checking if backuprepository's exist in %s", namespace)
+
+	backupRepos, err := GetBackupRepositoryList(c, namespace)
+	if err != nil {
+		return fmt.Errorf("failed to get BackupRepository list: %v", err)
```

Comment:
would check if error was not found error, and it is, return nil (there are no BackupRepositories in this namespace, nothing to delete), also keeping generic error handling as it is now

Reply:
fixed, thanks

Reply:
I think this will break tests, first run will always fail, because there is no prior backupRepo, and and error will be returned

need to check if the error is not found error here for cases where there is no prior backupRepo,no?

Reply:
I had the same worry as you are expressing.

on line 150 I have
backupRepositoryList := &velero.BackupRepositoryList{
		Items: []velero.BackupRepository{},

which should initialize the list w/ 0 items. 
Unless I'm misunderstanding something, GetBackupRepositoryList should at minimum return an empty list or a populated list.